src/FillRequestGroupCommand.cc

```diff
@@ -80,6 +80,17 @@ bool FillRequestGroupCommand::execute()
     }
   }
   e_->addRoutineCommand(std::unique_ptr<Command>(this));
+
+  // let's make sure we come back here every second or so
+  // if we use the optimize-concurrent-download option
+  if(rgman->getOptimizeSimultaneousDownloads()) {
+    const Timer now;
```

Comment:
If possible, please use global::wallclock(), which saves many syscalls to get time stamp.